### ISSUE TITLE
Device won't stop glitching out even in non-jailbroken state

### ISSUE THREAD

**Issue by andrewwiik** [kpwn/yalu102#374]
Hello qwerty,

I woke up to my screen glitching out all crazy (never touched custom screen resolution tweaks) so I decided I'll just reboot... still persists... okay... maybe on more reboot.... still happens when in unjailbroken mode.... okay lets erase all content and settibgs..... nope is still happening. I know this isn't a hardware issue with my screen btw.

**Comment by kpwn** [comment]
This is a pretty awful bug report, and have absolutely no clue what you might be talking about.

**Comment by andrewwiik** [comment]
I'm sorry I can't really provide more information. I'm trying to get a picture now of it to upload, it is less glitchy in apps but SpringBoard is totally fucked

**Comment by terry1994** [comment]
You should probably elaborate on some things:
Does it occur when the device is booting, on the boot screen (Apple logo)?
Have you checked what it may appear like through Quicktime display mirroring?
Does it respond to input?
How do you know it's not a hardware issue?

AFAIK, no permanent changes are really made, at least none that would actually be active when booting in a non-jailbroken state.

**Comment by andrewwiik** [comment]
![wg5c05k7](https://cloud.githubusercontent.com/assets/11397337/22549032/4f932406-e917-11e6-8800-78a136c32062.jpg)

**Comment by OGKevin** [comment]
😂😂👌🏾👌🏾 "pretty awful bug report "

**Comment by andrewwiik** [comment]
Happens always now, can't be in any state with out it

**Comment by kpwn** [comment]
Looks like your screen is toast.

**Comment by andrewwiik** [comment]
It is better in certain apps which is how i know it isn't the screen

**Comment by andrewwiik** [comment]
and before it starts loading the system and is just in iBoot the screen is also fine

**Comment by kpwn** [comment]
Certain apps and iBoot may use the screen differently. 

Your GPU may be toast, for instance. I am very positive this is a hardware bug, but I will try to make a testcase for you to try. Do you have aptickets saved for 10.2?

**Comment by OGKevin** [comment]
Can u provide a pick of when ur in an app?

**Comment by andrewwiik** [comment]
by APTickets do you mean blobs?

**Comment by andrewwiik** [comment]
![e1qcfiq9 1](https://cloud.githubusercontent.com/assets/11397337/22549172/085ac89a-e918-11e6-9cc1-b45374a6d0be.jpg)

**Comment by andrewwiik** [comment]
this is what i mean about better in certain apps @kpwn

**Comment by kpwn** [comment]
By aptickets I do mean blobs.


That screen looks awfully like hardware corruption by the way, and it's even clearer in this screenshot (notice the vertical lines? well turns out making a vertical line is not easy for memory corruption, since screen orders pixel left to right overflowing in the next line. I find it hard to believe that this is software induced, although it could be.

**Comment by andrewwiik** [comment]
Um yes I do I believe, I had someone save them for me, let me see if I can find them hold on

**Comment by kpwn** [comment]
write your generator to nvram and restore 10.2 and see if issue persists.

**Comment by andrewwiik** [comment]
okay

**Comment by OGKevin** [comment]
I also think this is hardware, did u try quicktime mirroring like @terrry1994 said. I think the mirror may look just fine.

**Comment by andrewwiik** [comment]
@OGKevin @terry1994 QuickTime isn't giving me the option of my phone as a screen, only mic

**Comment by titelheld** [comment]
@andrewwiik start quicktime, File > New Movie Recording > recording window should open, next to the record button select iPhone from the dropdown menu.

**Comment by andrewwiik** [comment]
I know, it only has iPhone for microphone

**Comment by TheMoleCZ** [comment]
A similar thing has once happened to me when I was jailbroken on iOS 6 or 7. Vertical lines and also everything was displaying only in the left corner of the screen, the rest was either lines or black. Everything was fine after I restored the device so I think this could also be a software issue.

**Comment by OGKevin** [comment]
Hmm, ill need to do some research. I thought if it was a partial hardware malfunction it should still be able to mirror. So if my initial thought is wrong, then  this means that without a good functioning GPU u  can't use screen  mirroring.

**Comment by OGKevin** [comment]
If you have Blobs saved u can indeed try and restore as it wont matter anyway if it was a hardware issue, meaning it will still be present after restore. If its software based then it should be indeed fixed after a restore.

**Comment by andrewwiik** [comment]
Going through the entire guide lol, takes a long time

**Comment by andrewwiik** [comment]
Request URL set to https://gs.apple.com/TSS/controller?action=2
TSS server returned: STATUS=94&MESSAGE=This device isn't eligible for the requested build.
ERROR: TSS request failed (status=94, message=This device isn't eligible for the requested build.)
Sending TSS request attempt 1... [Error] sep firmware isn't signed
Failed with errorcode=-3

**Comment by andrewwiik** [comment]
Any ideas?

**Comment by andrewwiik** [comment]
@kpwn

**Comment by PwnK** [comment]
> [Error] sep firmware isn't signed
You need to use ios 10.2.1's sep

**Comment by andrewwiik** [comment]
how

**Comment by PwnK** [comment]
Just take baseband and sep from ios 10.2.1's ipws

**Comment by andrewwiik** [comment]
didn't work? am i getting something wrong? just copy the two files?

**Comment by Aerozolic** [comment]
@andrewwiik use this guide: http://www.ipodhacks142.com/how-to-restore-to-ios-10-2-unsigned-using-prometheus-on-iphone-ipod-touch-or-ipad/

if you don't have openssh installed you might want to use [https://www.reddit.com/r/jailbreak/comments/5r2mwr/tutorial_how_to_use_dropbear_ssh_via_usb_on/?st=iyogmndz&sh=396a552b]this guide for ssh instead.

**Comment by andrewwiik** [comment]
I am, I get the baseband and SEp files from 10.2.1 IPSw and placed them in the Downgrade folder but get that error

**Comment by andrewwiik** [comment]
WIll have results on whether restore worked in about 10 minutes I believe @kpwn

**Comment by andrewwiik** [comment]
@kpwn I did a restore using prometheus and I still have the problem, looks like I'm taking a trip to the apple store to buy a new one, now the trouble of finding one on 10.2

**Comment by Aaahh** [comment]
@andrewwiik
Did you try a simple screenshot?
Happened to my friends phone and its stock
Easiest fix would be a new phone

**Comment by OGKevin** [comment]
U can get a screen replacement without them updating your 📱.

**Comment by andrewwiik** [comment]
The screen isn't the issue @OGKevin It recieves touches fine and if I screenshot it shows the correctView for a split second

**Comment by OGKevin** [comment]
oh yea true, it is your GPU not sure if they will replace that indeed.

**Comment by andrewwiik** [comment]
@kpwn sorry for this question and you can DM me @andywiik on twitter with the answer if you don't wanna share publicly but when can I anticipate the i7 10.1.1 supported version? I'm asking because I'm debating what device to buy outright if they rep is willing to go through serial numbers

**Comment by andrewwiik** [comment]
@kpwn I have concluded Yalu was not at fault, you may close this issue, thanks for your time

**Comment by kpwn** [comment]
I am curious about what was the cause behind this, by the way.

**Comment by andrewwiik** [comment]
I think the GPU is bad in someway

**Comment by andrewwiik** [comment]
@kpwn I wanted to give you an update along with my conclusion on this whole thing.

So all in all I was able to get phone to work again which meant it was a software thing. But I wanna explain what I think it was and see if you agree.

So when substrate safe-mode was broken I would use the set "Display Zoom" setting in Settings under "Display & Brightness". This "Zoom Mode" is the same way tweaks like Upscale and Littlebrother work. A little over a year ago I was beta testing Upscale on a old iPod 5 and when you set the resolution too high the screen would start to tear and stuff. I now remember it looked very similar to what I had now. I had assumed i had fixed the glitching on my iPhone 6 by pressing on a certain spot on the screen, but that wasn't it... It was right after i had finally been able to get past setting the display zoom after restore. I believe my abusive use of setting display zoom as a easy respring caused it's values to somehow become corrupt, and they finally got corrected after i had gotten past that step in setup.
